aws/data_source_aws_vpc_endpoint_service.go

```diff
@@ -76,13 +76,14 @@ func dataSourceAwsVpcEndpointServiceRead(d *schema.ResourceData, meta interface{
 	conn := meta.(*AWSClient).ec2conn
 
 	var serviceName string
-	if v, ok := d.GetOk("service_name"); ok {
-		serviceName = v.(string)
-	} else if v, ok := d.GetOk("service"); ok {
-		serviceName = fmt.Sprintf("com.amazonaws.%s.%s", meta.(*AWSClient).region, v.(string))
-	} else {
-		return fmt.Errorf(
-			"One of ['service', 'service_name'] must be set to query VPC Endpoint Services")
+	if v := d.Get("service_name").(string); v != "" {
+		serviceName = v
+	} else if v := d.Get("service"); v != "" {
+		if v == "sagemaker.notebook" {
+			serviceName = fmt.Sprintf("aws.sagemaker.%s.notebook", meta.(*AWSClient).region)
+		} else {
+			serviceName = fmt.Sprintf("com.amazonaws.%s.%s", meta.(*AWSClient).region, v)
+		}
```

Comment:
Hey @ewbankkit 👋 While supporting `sagemaker.notebook` is a nice enhancement, it does not cover the original bug report configuration:

```hcl
data "aws_vpc_endpoint_service" "sagemaker_notebook" {
  service = "aws.sagemaker.us-east-1.notebook"
}
```

It seems like we may want to allow arbitrary `service` configurations to pass through as-is if they contain multiple periods (or some other loose heuristic) so we don't need to special case a bunch of values should other service teams choose their own standards 🙃.

Reply:
Yes, this is a tough one.
The `service_name` field is designed to allow the full reverse-DNS name of a service to be specified and `service = "sagemaker.runtime"` for example just happens to work currently as the AWS service ID is `runtime.sagemaker`.
Let me think about this for a while.
Specifying `service = "aws.sagemaker.us-east-1.notebook"` is the workaround for the specific problem.

Reply:
I think the best thing here may be to have no code changes and tighten up the documentation for both the data source and corresponding resource. The underlying AWS API has no notion of just `service` (it deals with `ServiceName`) so this argument is really just for convenience.